src/msg/async/rdma/RDMAStack.h

```diff
@@ -235,6 +237,20 @@ class RDMAConnectedSocketImpl : public ConnectedSocketImpl {
       active = false;
     }
   };
+  class C_handle_connection_init : public EventCallback {
+    RDMAConnectedSocketImpl *csi;
+    bool active;
+   public:
+    C_handle_connection_init(RDMAConnectedSocketImpl *w) : csi(w), active(true) {}
+    void do_request(uint64_t fd) {
+      if (active)
+        csi->handle_connection_init();
```

Comment:
why fault the connection?

Reply:
if this conneciton is connected,  the function handle_connection_init should't be called,  except some logic error , such as the con_init_handler is not deleted from the event center. 
if we failed to send_msg,  I think we need to fault the connection.

Reply:
i think we'd better add an assert here like
```c++
if (active) {
  assert(!csi->is_connected());
  csi->handle_connection_established();
```

Reply:
> i think we'd better add an assert here like
> 
> ```c++
> if (active) {
>   assert(!csi->is_connected());
>   csi->handle_connection_established();
> ```

we should not call `assert(!csi->is_connected())` in here,  `handle_connection_established()` will check the connection status.